File: e2e-simulation/test-500.py
import httpx
import asyncio

async def test_api():
    payload = {
        "events": [
            {
                "id": "e2e-test-safe-1234",
                "type": "page_scan",
                "org_id": "org_default",
                "user_id": 29, 
                "device_id": "device-1",
                "url": "https://example.com",
                "domain": "example.com",
                "risk_score": 5,
                "verdict": "allow",
                "details": {"decision": "allow", "module": "url_model", "scan_type": "navigation"}
            }
        ]
    }
    
    async with httpx.AsyncClient() as client:
        try:
            resp = await client.post("http://localhost:8000/events/ingest", json=payload)
            print(f"Status: {resp.status_code}")
            print(f"Response: {resp.text}")
        except Exception as e:
            print(f"Exception: {e}")

if __name__ == "__main__":
    asyncio.run(test_api())

File: e2e-simulation/test-500.ts
import axios from 'axios';
import { Config } from './config';
import { generateEmployees } from './fixtures/profiles';

async function test() {
  try {
    const employees = generateEmployees('org_default', 5, '20260831', 'admin@example.com');
    const emp = employees.find(e => e.actorType === 'api_synthetic');
    
    const events = [{
      id: `e2e-${emp.id}-test`,
      type: 'page_scan',
      org_id: 'org_default',
      device_id: 'test-device',
      url: 'https://github.com',
      domain: 'github.com',
      risk_score: 0,
      verdict: 'allow',
      details: { decision: 'allow' },
      user_id: undefined
    }];

    console.log("Sending payload:", JSON.stringify({ events }, null, 2));

    const res = await axios.post('http://localhost:8000/events/ingest', { events });
    console.log("SUCCESS:", res.data);
  } catch (err) {
    console.error("FAILED:");
    if (err.response) {
      console.error(err.response.status, err.response.data);
    } else {
      console.error(err.message);
    }
  }
}
test();
